src/EditorFeatures/Test2/Formatting/FormattingCommandHandlerTests.vb

```diff
@@ -0,0 +1,65 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
+    ''' <summary>
+    ''' Tests that want to exercise as much of the real command handling stack as possible.
+    ''' </summary>
+    <[UseExportProvider]>
+    Public Class FormattingCommandHandlerTests
+        Public Shared ReadOnly Property AllCompletionImplementations() As IEnumerable(Of Object()) =
+            TestStateFactory.GetAllCompletionImplementations()
+
+        <WorkItem(912965, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/912965")>
+        <MemberData(NameOf(AllCompletionImplementations))>
+        <WpfTheory, Trait(Traits.Feature, Traits.Features.Formatting)>
+        Public Sub TypingUsingStatementsProperlyAligns1(completionImplementation As CompletionImplementation)
+            Using state = TestStateFactory.CreateCSharpTestState(completionImplementation,
+                              <Document>
+using System;
+class TestClass
+{
+    void TestMethod(IDisposable obj)
+    {
+        $$
+    }
+}
+                              </Document>, includeFormatCommandHandler:=True)
+                state.SendTypeChars("using (obj)")
+                state.SendReturn()
+
+                ' we should be indented one level from the using
+                AssertVirtualCaretColumn(state, 12)
+
+                ' We should continue being indentded one level in
+                state.SendTypeChars("u")
+                Assert.Equal("            u", state.GetLineTextFromCaretPosition())
+
+                ' Until close paren is typed.  Then, we should align the usings
```

Comment:
💡 This would benefit from typing `sing (obj` first, and then typing `)` by itself

Reply:
Sure.  if you prefer that!

Reply:
done.

Reply:
Done!